### ISSUE TITLE
Generic interpolators

### ISSUE THREAD

**Issue by jbrezmorf** [flow123d/flow123d#823]
Extension of #774.

Make separate class or method that creates or initialize FieldFE (or Field containing FieldFE) from other Field. Use cases:
- interpolation of init conditions to correct FE space
- interpolation between meshes
- interpolation as caching of FieldFormula data
- interpolation of output FE fields of one equation to natural FE fields of other equation.

**Comment by jbrezmorf** [comment]
## Generic interpolation  and vectorized assembly

- no problem with interpolation as a separate assembly step followed possibly by a system solution (L2 assembly)
- however this may be prohibitive step
    - memory access limited
    - storing full interpolated DOF vector
    - makes a synchronization point in parallel calculations
- would be nice to have support for local interpolation calculated on the fly, i.e. interpolated field may be evaluated just by evaluating the source field in some DOFs and multiplying by a suitable matrix to compute the new DOFs 
- there is a problem, that this interpolation may introduce a new quadrature/evaluation points. The assembly algorithm must be extended to this end.

**Comment by jbrezmorf** [comment]
## Generic interpolation  and vectorized assembly

- no problem with interpolation as a separate assembly step followed possibly by a system solution (L2 assembly)
- however this may be a prohibitive step
    - memory access limited
    - storing full interpolated DOF vector
    - makes a synchronization point in parallel calculations
- would be nice to have support for local interpolation calculated on the fly, i.e. interpolated field may be evaluated just by evaluating the source field in some DOFs and multiplying by a suitable matrix to compute the new DOFs 
- there is a problem, that this interpolation may introduce new quadrature/evaluation points. The assembly algorithm must be extended to this end.
